src/magic.c

```diff
@@ -1252,6 +1252,24 @@ TbResult magic_use_power_heal(PlayerNumber plyr_idx, struct Thing *thing, MapSub
     powerst = get_power_model_stats(PwrK_HEALCRTR);
     thing_play_sample(thing, powerst->select_sound_idx, NORMAL_PITCH, 0, 3, 0, 2, FULL_LOUDNESS);
     apply_spell_effect_to_thing(thing, SplK_Heal, splevel);
+    if (gameadd.heal_radius)
+    {
+        unsigned char radius = splevel + 1;
+        for (MapSubtlCoord stl_x = (thing->mappos.x.stl.num - (radius >> 1)); stl_x <= (thing->mappos.x.stl.num + (radius >> 1)); stl_x++)
+        {
+            for (MapSubtlCoord stl_y = (thing->mappos.y.stl.num - (radius >> 1)); stl_y <= (thing->mappos.y.stl.num + (radius >> 1)); stl_y++)
+            {
```

Comment:
Maybe instead use some scale factor? like `if (dist(thing, cast_point) <  gameadd.heal_radius * splevel)`

Reply:
gameadd.heal_radius is a boolean.

Reply:
I see it. But why?
Now a lvl10 dragon would heal 5 stl around (almost two slabs)

Reply:
The manual says the radius was determined by the spell level. I'm not sure if it's supposed to apply to the creature spell too?

Reply:
Ouch. We are talking about keeper power spell. 
So I think it should have reduced area effect if any at all. 
Now Players and AIs cast full scaled heal at each hurt creature in combat for 1200gp per cast.
Such change reduce price of such healing by number of fighting creatures nearby. 
Lets estimate this around 4. It is too much.

Reply:
Well, the manual does say of the creature spell: 'Also, any nearby creatures are healed by a quarter', but I'm not sure it's possible to do that directly since we're dealing with spell levels rather than actual health amounts (not to mention it doesn't say how 'nearby' is).

Am I right in assuming that spell levels start from 0?